### ISSUE TITLE
Not Able to consume data from hashicorp vault version kv engine

### ISSUE THREAD

**Issue by 2dev2** [tellerops/teller#115]
## Expected Behavior
We have got the secret data from hashicorp vault HTTP API through postman, teller also should be able to consume the same data.

## Current Behavior
For same data, its not able to consume the data and giving panic: runtime error: invalid memory address or nil pointer dereference

## Possible Solution
We need to somehow pass some key from config which field we want to decode from response.
From Curl we can get secret data like `curl | jq -r '.data.data'. `which is a map expected by teller

## Steps to Reproduce
1. Create a new cluster in hashicorp cloud  with versioned KV2 engine enabled.
2. Store some secret key by creating a namespace and path.
3. Access the Same secret key data through curl, we will be able to get the data
4. Pass same URL path in teller config and try to access the data.

##Context
Response Data from API got changed and due to this  getting error. This can happen with any provider, so we need to pass some key which we want to consume in our teller application for creating map.

## Specifications
  - Version: `Teller 1.5.2 Revision 8c37a9df22f2f64ba10f71247cc8bcdf95f4ff51, date: 2022-06-21T06:34:51Z`
  - Platform: MACOS 
<img width="772" alt="Screenshot 2022-06-26 at 6 59 04 PM" src="https://user-images.githubusercontent.com/8681544/175816854-99870897-2a92-4ac3-9975-8aef232f7cd2.png">
<img width="1741" alt="Screenshot 2022-06-26 at 6 57 16 PM" src="https://user-images.githubusercontent.com/8681544/175816857-b842486d-f1bf-4903-bc99-3309bef16b44.png">
<img width="1403" alt="Screenshot 2022-06-26 at 6 55 22 PM" src="https://user-images.githubusercontent.com/8681544/175816861-06ff6dc6-64e8-475f-8ed5-a2895a09da11.png">

**Comment by kaplanelad** [comment]
Hey @2dev2 thanks for reporting this issue, i'll check it

**Comment by 2dev2** [comment]
Do i am missing something, please correct me if its not bug.

**Comment by kaplanelad** [comment]
Hey @2dev2, thanks for reporting this issue. i found a bug where we return secret and fixed it in pr #120 
But didn't succeed in reproducing your case. I suspect the path you added in `.teller.yaml` is incorrect.
see my following steps to check your issue (you can check is as well on clean vault)

### Run Vault locally
```
$ vault --version
$ Vault v1.11.0 ('ea296ccf58507b25051bc0597379c467046eb2f1+CHANGES'), built 2022-06-17T15:48:44Z
$ vault server -dev
$ export VAULT_ADDR='http://127.0.0.1:8200'
$ export VAULT_TOKEN=xxx
``` 

### Create a secret json file
```sh
{
  "options": {
    "cas": 0
  },
  "data": {
    "foo": "bar",
    "secret-version": "test"
  }
}
```

### Put secret:
```sh
curl \
    --header "X-Vault-Token: <SECRET>" \
    --request POST \
    --data @payload.json \
    http://127.0.0.1:8200/v1/secret/data/my-kv-secrets/test/demo
```

### .teller.yaml file
```yaml
project: vault
providers:
  hashicorp_vault:
    env_sync:
      path: secret/data/my-kv-secrets/test/demo
```

### Run telelr
```sh
teller show
```

**Comment by kaplanelad** [comment]
@2dev2, the panic releasee in v1.5.3.
